### PR TITLE
feat: handle `UnaryOpsWithBigInt`

### LINKED ISSUE #454: Port `testUnaryOpsWithBigInt`
Port [testUnaryOpsWithBigInt](https://github.com/google/closure-compiler/blob/a4c880032fba961f7a6c06ef99daa3641810bfdd/test/com/google/javascript/jscomp/PeepholeFoldConstantsTest.java#L590).

[These cases](https://github.com/Boshen/oxc/blob/de9d4de27670276cbc66f7064a173be58c141145/crates/oxc_minifier/tests/closure/fold_constants.rs#LL69C1-L74C2) have been added. You may want to start coding from [`try_fold_unary_operator`](https://github.com/Boshen/oxc/blob/de9d4de27670276cbc66f7064a173be58c141145/crates/oxc_minifier/src/compressor/fold.rs#LL271C1-L272C1).

Currently, `-1n` will be parsed to:

```bash
{
    "type": "ExpressionStatement",
    "start": 0,
    "end": 3,
    "expression": {
      "type": "UnaryExpression",
      "start": 0,
      "end": 3,
      "operator": "-",
      "argument": {
        "type": "BigintLiteral",
        "start": 1,
        "end": 3,
        "value": "1n"
      }
    }
}
```

we need to fold it into the following struct:

```bash
{
    "type": "ExpressionStatement",
    "start": 0,
    "end": 3,
    "expression": {
        "type": "BigintLiteral",
        "start": 0,
        "end": 3,
        "value": "-1n"
    }
}
```

For support negative value in our [`BigIntLiteral`](https://github.com/Boshen/oxc/blob/de9d4de27670276cbc66f7064a173be58c141145/crates/oxc_hir/src/hir.rs#L285), we can change it to:

```diff
use num_bigint::{BigInt};

#[derive(Debug, Clone, Hash)]
#[cfg_attr(feature = "serde", derive(Serialize), serde(tag = "type"))]
pub struct BigintLiteral {
    #[cfg_attr(feature = "serde", serde(flatten))]
    pub span: Span,
    #[cfg_attr(feature = "serde", serde(serialize_with = "crate::serialize::serialize_bigint"))]
-   pub value: BigUint,
+   pub value: BigInt,
}
```

### PR DESCRIPTION
**Pull request by cijiugechu**
close #454

### CODE REVIEW

**Comment by Boshen** [comment]
Nice work! Conformance test is failing, you can run `just c` to update the snapshots. It is currently failing on language/expressions/subtraction/bigint-arithmetic.js https://github.com/tc39/test262/blob/main/test/language/expressions/subtraction/bigint-arithmetic.js

**Comment by Boshen** [comment]
You need to find where it is failing and fix the code. A good way to find this is to copy the file and use `crates/oxc_minifier/examples/minifier.rs` to find the broken code.

**Comment by Boshen** [comment]
@mysteryven PTAL and merge if everything is ok.

File: crates/oxc_minifier/src/compressor/fold.rs [hunk 1]
```diff
@@ -296,19 +301,27 @@ impl<'a> Compressor<'a> {
                         let bool_literal = self.hir.boolean_literal(unary_expr.span, !boolean);
                         return Some(self.hir.literal_boolean_expression(bool_literal));
                     }
-                }
+                    Expression::BigintLiteral(big_int_literal) => {
+                        if big_int_literal.value.is_zero() {
```
Comment (mysteryven) [inline crates/oxc_minifier/src/compressor/fold.rs:305]: Maybe we don't need this check. 🤔 `!0n` can be folded to `true`, and then will be compressed to `!0` in [`compress_boolean`](https://github.com/Boshen/oxc/blob/b8f5e3ad929a041c6b980b139bebe86fb398729a/crates/oxc_minifier/src/compressor/mod.rs#L286). Which is shorter.

**Comment by mysteryven** [comment]
Thanks for your great work! If you are interested in any pieces of [`FoldConstants`](https://github.com/google/closure-compiler/blob/master/test/com/google/javascript/jscomp/PeepholeFoldConstantsTest.java), just leave a comment. 😀

**Comment by cijiugechu** [comment]
> Thanks for your great work! If you are interested in any pieces of [`FoldConstants`](https://github.com/google/closure-compiler/blob/master/test/com/google/javascript/jscomp/PeepholeFoldConstantsTest.java), just leave a comment. 😀

Of course, I could probably start with this #453 😊

**Pull request merged**